ebpf/src/processeventssubscriber.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2019-present Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "processeventssubscriber.h"
+#include "probes/kprobe_group/header.h"
+
+#include <asm/unistd_64.h>
+
+namespace trailofbits {
+namespace {
+// clang-format off
+const std::unordered_map<std::uint64_t, std::string> kSyscallNameTable = {
+  { __NR_execve, "execve" },
+  { __NR_execveat, "execveat" },
+  { KPROBE_FORK_CALL, "fork" },
+  { KPROBE_VFORK_CALL, "vfork" },
+  { KPROBE_CLONE_CALL, "clone" }
+};
+// clang-format on
+
+const std::string& getSystemCallName(std::uint64_t system_call_nr) {
+  static const std::string kUnknownSystemCallName{"unknown"};
+
+  auto name_it = kSyscallNameTable.find(system_call_nr);
+  if (name_it == kSyscallNameTable.end()) {
+    return kUnknownSystemCallName;
+  }
+
+  return name_it->second;
+}
+} // namespace
+
+// clang-format off
+BEGIN_TABLE(ebpf_process_events)
+  TABLE_COLUMN(timestamp, osquery::TEXT_TYPE)
+  TABLE_COLUMN(ppid, osquery::TEXT_TYPE)
```

Comment:
Can these be integral types instead of text? Would probably save a few copies/allocations below.